waspc/src/Wasp/Generator.hs

```diff
@@ -23,21 +23,37 @@ import qualified Wasp.Generator.ServerGenerator as ServerGenerator
 import qualified Wasp.Generator.Setup
 import qualified Wasp.Generator.Start
 import Wasp.Generator.WebAppGenerator (generateWebApp)
+import Wasp.GeneratorMonad (Generator, GeneratorError, GeneratorState (warnings), GeneratorWarning, runGenerator)
+import Wasp.Util ((<++>))
 
 -- | Generates web app code from given Wasp and writes it to given destination directory.
 --   If dstDir does not exist yet, it will be created.
 --   NOTE(martin): What if there is already smth in the dstDir? It is probably best
 --     if we clean it up first? But we don't want this to end up with us deleting stuff
 --     from user's machine. Maybe we just overwrite and we are good?
-writeWebAppCode :: AppSpec -> Path' Abs (Dir ProjectRootDir) -> IO ()
+writeWebAppCode :: AppSpec -> Path' Abs (Dir ProjectRootDir) -> IO (Either GeneratorError [GeneratorWarning])
```

Comment:
Yeah, thanks for the discussion on this one. It was one of the trickiest parts for me to work out. I do think we can use `IO ([GeneratorWarning], [GeneratorError])` as well, if we assume as you noted that the presence of _any_ error meant the generation failed.

The only case that goes against this I was unsure about is if there was an error thrown somewhere down the stack, but we caught it and then continued to carry on in some satisfactory way. In that case, it would be logged (and thus in the list of errors), but perhaps we recovered somehow and saved the generation to still be considered a success? That was the thinking with `Either`, but I am not sure just how that can happen yet, so maybe we just use the simple case and expand if that case becomes a reality. 👍🏻  Thanks

Reply:
Hm that is a very good point! The question is then, are there "non fatal" errors?

I think it is best to continue as you concluded at the end -> let's assume that errors are fatal, and while generator can try to collect multiple of them to report them, as long as one of them happens, we can consider that generator has failed. That way we have a relatively simple situation here. And warnings are here for "non-fatal" stuff.

Reply:
That sounds good to me! 👍🏻

Reply:
Updated to: `writeWebAppCode :: AppSpec -> Path' Abs (Dir ProjectRootDir) -> IO ([GeneratorWarning], [GeneratorError])`